fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordContextProperties.java

```diff
@@ -99,6 +99,10 @@ public final class LuceneRecordContextProperties {
      * Number of documents to move from a partition when its size exceeds {@link com.apple.foundationdb.record.lucene.LuceneIndexOptions#INDEX_PARTITION_HIGH_WATERMARK}.
      */
     public static final RecordLayerPropertyKey<Integer> LUCENE_REPARTITION_DOCUMENT_COUNT = RecordLayerPropertyKey.integerPropertyKey("com.apple.foundationdb.record.lucene.repartition.document.count", 16);
+    /**
+     * Maximum number of documents to move during a re-balancing run.
+     */
+    public static final RecordLayerPropertyKey<Integer> LUCENE_REPARTITION_MAX_DOCUMENT_COUNT = RecordLayerPropertyKey.integerPropertyKey("com.apple.foundationdb.record.lucene.repartition.max.document.count", 1000);
```

Comment:
Would `LUCENE_MAX_DOCUMENTS_TO_MOVE_DURING_REBALANCE` (or something similar) be more descriptive?